test_harness/src/lib.rs

```diff
@@ -884,6 +900,396 @@ mod tests {
+                &UnifiedNum::from_u64(54000),
+                "Verifying correct spender balance for campaign creator"
+            );
+            assert_eq!(
+                newstate
+                    .balances
+                    .earners
+                    .get(&CAMPAIGN_1.channel.leader.to_address())
+                    .expect("Should retrieve earner balance"),
+                &UnifiedNum::from_u64(27000),
+                "Verifying correct earner balance for leader"
+            );
+            assert_eq!(
+                newstate
+                    .balances
+                    .earners
+                    .get(&CAMPAIGN_1.channel.follower.to_address())
+                    .expect("Should retrieve earner balance"),
+                &UnifiedNum::from_u64(18000),
+                "Verifying correct earner balance for follower"
+            );
+            assert_eq!(
+                newstate
+                    .balances
+                    .earners
+                    .get(&*PUBLISHER)
+                    .expect("Should retrieve earner balance"),
+                &UnifiedNum::from_u64(9000),
+                "Verifying correct earner balance for publisher"
+            );
+
+            // Balances are being changed since the last propagated message ensuring that a new NewState will be generated
+            accounting_balances
+                .spend(campaign.creator, *PUBLISHER, UnifiedNum::from(9_000))
+                .expect("Should spend for Publisher");
+            let new_state = get_new_state_msg(
+                &leader_sentry,
+                &accounting_balances,
+                contracts.token.0.precision.get(),
+            );
+
+            // Propagating a new NewState so that we can verify it updates properly
+            leader_sentry_with_propagate
+                .propagate(new_channel, &[&MessageTypes::NewState(new_state.clone())])
+                .await;
+            follower_sentry_with_propagate
+                .propagate(new_channel, &[&MessageTypes::NewState(new_state)])
+                .await;
+
+            let newstate = leader_sentry_with_propagate
+                .get_our_latest_msg(new_channel.id(), &["NewState"])
+                .await
+                .expect("should fetch")
+                .unwrap();
+
+            let newstate_follower = follower_sentry_with_propagate
+                .get_our_latest_msg(new_channel.id(), &["NewState"])
+                .await
+                .expect("should fetch")
+                .unwrap();
+
+            let newstate = NewState::<CheckedState>::try_from(newstate).expect("Should convert");
+            let newstate_follower =
+                NewState::<CheckedState>::try_from(newstate_follower).expect("Should convert");
+
+            assert_eq!(
+                newstate.state_root, newstate_follower.state_root,
+                "Stateroots of the new messages match"
+            );
```

Comment:
It does make some sense to test if the propagated NewState is returned, however, it would be better to submit new events and expect a NewState returned with the correct payouts instead.